src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/TravellerSelection.tsx

```diff
@@ -27,65 +33,141 @@ export function TravellerSelection({
   selectionMode,
   fareProductType,
 }: TravellerSelectionProps) {
-  const {t} = useTranslation();
+  const {t, language} = useTranslation();
+  const styles = useStyles();
+  const {open: openBottomSheet, close: closeBottomSheet} = useBottomSheet();
 
-  const userCountState = useUserCountState(selectableUserProfiles);
-  const selectableUserProfilesWithCount =
-    userCountState.userProfilesWithCount.filter((a) =>
-      selectableUserProfiles.some((b) => a.id === b.id),
-    );
-  useEffect(() => {
-    if (selectionMode == 'none' && selectableUserProfiles.length > 0) {
-      // If selectionMode is none, we should default to the first item.
-      // selectableUserProfiles should be only have one item, but this
-      // is specified by the fare product and reference data in firestore.
-      setTravellerSelection([
-        {
-          ...selectableUserProfiles[0],
-          count: 1,
-        },
-      ]);
-    } else {
-      const filteredSelection = userCountState.userProfilesWithCount.filter(
-        (u) => selectableUserProfiles.find((i) => i.id === u.id),
-      );
-      setTravellerSelection(filteredSelection);
-    }
-  }, [userCountState.userProfilesWithCount, fareProductType, selectionMode]);
+  const accessibility: AccessibilityProps = {
+    accessible: true,
+    accessibilityRole: 'button',
+    accessibilityLabel:
+      'Todo: Some label in the correct language here' + screenReaderPause,
+    accessibilityHint: t(PurchaseOverviewTexts.zones.a11yHint),
+  };
+
+  const [selectableUserProfilesWithCount, setSelectableUserProfilesWithCount] =
+    useState<UserProfileWithCount[]>(selectableUserProfiles);
 
   useEffect(() => {
-    userCountState.updateSelectable(selectableUserProfiles);
-  }, [selectableUserProfiles]);
+    const filteredSelection = selectableUserProfilesWithCount.filter((u) =>
+      selectableUserProfiles.find((i) => i.id === u.id),
+    );
+    setTravellerSelection(filteredSelection);
+  }, [fareProductType, selectionMode, selectableUserProfilesWithCount]);
 
   if (selectionMode === 'none') {
     return <></>;
   }
 
+  let numberOfPassengerCategoriesSelectedFrom = 0;
+  const totalPassengersCount = selectableUserProfilesWithCount.reduce(
+    (acc, sUPWC) => {
+      if (sUPWC.count > 0) {
+        numberOfPassengerCategoriesSelectedFrom += 1;
```

Comment:
I don't like side effects in `reduce` or other functional concepts, since they easily go under the radar.

I would rather like something like this:
```ts
const numberOfPassengerCategoriesSelectedFrom = 
  selectableUserProfilesWithCount.filter(u => u.count).length
  ```

Reply:
Fair point!